### PR TITLE
[RNMobile] Upgrade compile and target sdk version to Android API 31

### PR DESCRIPTION
**Pull request by fluiddot**
Related PRs:
- **Parent PR in WordPress-Android:** https://github.com/wordpress-mobile/WordPress-Android/pull/17153
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5184
- https://github.com/wordpress-mobile/WordPress-Android/pull/17235

## What?

Upgrade `react-native-aztec`, `react-native-bridge` and `react-native-editor` projects to Android 12 (API 31).

## Why?

[Google Play requires apps to update to Android 12 (API level 31) before 31 November 1, 2022](https://support.google.com/googleplay/android-developer/answer/11926878?hl=en).

## How?

`compileSdkVersion` and `targetSdkVersion` properties of all Android projects have been updated to API 31.

Additionally, the main activity of the demo Android demo project is now exported. This is a behavior change required when using Android 12 related to safer component exporting ([reference](https://developer.android.com/about/versions/12/behavior-changes-12)).

The Android dependencies of forked repositories have been also updated to Android 12:
- [react-native-gesture-handler](https://github.com/wordpress-mobile/react-native-gesture-handler/pull/23)
- [react-native-reanimated](https://github.com/wordpress-mobile/react-native-reanimated/pull/19)
- [react-native-prompt-android](https://github.com/wordpress-mobile/react-native-prompt-android/pull/9)
- [react-native-linear-gradient](https://github.com/wordpress-mobile/react-native-linear-gradient/pull/7)
- [react-native-slider](https://github.com/wordpress-mobile/react-native-slider/pull/12)
- [react-native-video](https://github.com/wordpress-mobile/react-native-video/pull/20)

## Testing Instructions

This change doesn't affect a specific area, hence to test it we should basically smoke test the editor and check that works as expected.

## Screenshots or screencast 
N/A

### CODE REVIEW

**Comment by fluiddot** [comment]
I tested the demo project using a simulator and haven't found any issues. Looks like the editor is working nice with Android 12 🎊 .

File: packages/react-native-aztec/android/build.gradle [hunk 1]
```diff
@@ -7,9 +7,9 @@ buildscript {
         robolectricVersion = '3.5.1'
         jUnitVersion = '4.12'
         jSoupVersion = '1.10.3'
-        wordpressUtilsVersion = '2.3.0'
+        wordpressUtilsVersion = '110-13b4f06680ed8c145cbd09d63c4998dfe8b3b9de'
         espressoVersion = '3.0.1'
-        aztecVersion = 'v1.6.0'
+        aztecVersion = '1006-3fdbbc2086fb71d3750b01c013fa5cf6024c3142'
```
Comment (fluiddot) [inline packages/react-native-aztec/android/build.gradle]: The PRs that point these dependencies are the ones where Android API version is also being updated to 31:
- https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/110
- https://github.com/wordpress-mobile/AztecEditor-Android/pull/1006

File: packages/react-native-bridge/android/react-native-bridge/build.gradle [hunk 2]
```diff
@@ -65,20 +65,20 @@ dependencies {
...
-    implementation "com.github.wordpress-mobile:react-native-prompt-android:${extractPackageVersion(packageJson, 'react-native-prompt-android', 'dependencies')}"
+    implementation "com.github.wordpress-mobile:react-native-reanimated:update~android-12-1.9.0-g0ae9a17"
+    implementation "com.github.wordpress-mobile:react-native-prompt-android:update~android-12-SNAPSHOT"
     implementation "org.wordpress-mobile:react-native-webview:${extractPackageVersion(packageJson, 'react-native-webview', 'dependencies')}"
     implementation "org.wordpress-mobile:react-native-clipboard:${extractPackageVersion(packageJson, '@react-native-clipboard/clipboard', 'dependencies')}"
     implementation "org.wordpress-mobile:react-native-fast-image:${extractPackageVersion(packageJson, 'react-native-fast-image', 'dependencies')}"
 
-    implementation("com.github.wordpress-mobile:react-native-gesture-handler:${extractPackageVersion(packageJson, 'react-native-gesture-handler', 'dependencies')}", {
+    implementation("com.github.wordpress-mobile:react-native-gesture-handler:update~android-12-SNAPSHOT", {
```
Comment (fluiddot) [inline packages/react-native-bridge/android/react-native-bridge/build.gradle]: The reference of these dependencies will be updated once their associated PRs, which also update the Android API version are merged:
- [react-native-gesture-handler](https://github.com/wordpress-mobile/react-native-gesture-handler/pull/23)
- [react-native-reanimated](https://github.com/wordpress-mobile/react-native-reanimated/pull/19)
- [react-native-prompt-android](https://github.com/wordpress-mobile/react-native-prompt-android/pull/9)
- [react-native-linear-gradient](https://github.com/wordpress-mobile/react-native-linear-gradient/pull/7)
- [react-native-slider](https://github.com/wordpress-mobile/react-native-slider/pull/12)
- [react-native-video](https://github.com/wordpress-mobile/react-native-video/pull/20)

**Review by jhnstn — APPROVED** [review]
I tested the changes out via the demo app. I ran through all the flows that I could think of that might be impacted by the dependency changes. Everything worked as expected.
Looks good 👍

**Comment by fluiddot** [comment]
I noticed that the List block is not rendering the items properly (see attached screenshots). The interesting part is that I tried using the latest changes from Gutenberg `trunk` in combination with the WP-Android branch with Android 12 ([`issue/16068-android-12-update-target-sdk`](https://github.com/wordpress-mobile/WordPress-Android/pull/17153)) and still happens, so doesn't look like is related to Gutenberg code at first glance. We should investigate further the cause 🔍 .

Before|After
--|--
|

**Comment by fluiddot** [comment]
> I noticed that the List block is not rendering the items properly (see attached screenshots). The interesting part is that I tried using the latest changes from Gutenberg trunk in combination with the WP-Android branch with Android 12 (https://github.com/wordpress-mobile/WordPress-Android/pull/17153) and still happens, so doesn't look like is related to Gutenberg code at first glance. We should investigate further the cause 🔍 .

The issue is not related to Android 12 but using a newer version of `Aztec-Android`. Looks like the issue was introduced in [`v1.60.1`](https://github.com/wordpress-mobile/AztecEditor-Android/releases/tag/v1.6.1) as in [`v1.60.0`](https://github.com/wordpress-mobile/AztecEditor-Android/releases/tag/v1.6.0) wasn't happening. I'll investigate further the changes from https://github.com/wordpress-mobile/AztecEditor-Android/pull/992 as it's likely that affected styling on other blocks.

Aztec-Android `v1.60.0`|Aztec-Android `v1.60.1`
--|--
|

**Comment by fluiddot** [comment]
> The issue is not related to Android 12 but using a newer version of `Aztec-Android`. Looks like the issue was introduced in [`v1.60.1`](https://github.com/wordpress-mobile/AztecEditor-Android/releases/tag/v1.6.1) as in [`v1.60.0`](https://github.com/wordpress-mobile/AztecEditor-Android/releases/tag/v1.6.0) wasn't happening. I'll investigate further the changes from [wordpress-mobile/AztecEditor-Android#992](https://github.com/wordpress-mobile/AztecEditor-Android/pull/992) as it's likely that affected styling on other blocks.

After debugging the issue, I realized that is not a bug per se but caused by the introduction of a new dimension value ([reference](https://github.com/wordpress-mobile/AztecEditor-Android/pull/992/files#diff-89a37e2215d677870ce5dc923d25f75d94d3942b1f7b6d08490e46d01add2e40R24)) to control the vertical margin of blocks. By default the value is `8dp` which was causing the extra space.

This has been fixed in https://github.com/WordPress/gutenberg/pull/44610/commits/fdc7cc888f5d38a9260d94d768c657766fc8efff by simply setting the value to `0dp`.

File: packages/react-native-aztec/android/build.gradle (hunk 1 above)
Comment (fluiddot) [reply packages/react-native-aztec/android/build.gradle]: Solving this as Aztec now points to the latest version `v1.6.2`.

File: packages/react-native-bridge/android/react-native-bridge/build.gradle (hunk 2 above)
Comment (fluiddot) [reply packages/react-native-bridge/android/react-native-bridge/build.gradle]: Solving this as all dependencies have been updated.

**Pull request merged**